xonsh/environ.py

```diff
@@ -926,6 +928,15 @@ def _current_job():
     return ''
 
 
+def env_name(pre_chars='(', post_chars=') '):
+    """Extract the current environment name from $VIRTUAL_ENV."""
+    env_path = builtins.__xonsh_env__.get('VIRTUAL_ENV', '')
```

Comment:
I think that `VIRTUAL_ENV` should be properly handled and registered by the envrionment.  See the `xonsh.environ` file. Here you can register defaults, validators, and docstrings.  Then  this would get documentation automatically generated, you could use `__xonsh_env__.get('VIRTUAL_ENV')`, and you could ensure that this is always a string.